src/main/scala/workflow/WorkflowUtils.scala

```diff
@@ -0,0 +1,124 @@
+package workflow
+
+import scala.collection.mutable.ArrayBuffer
+
+
+object WorkflowUtils {
+  def instructionsToPipeline[A, B](instructions: Seq[Instruction]): Pipeline[A, B] = {
```

Comment:
It is not exactly fully deterministic and symmetric, but I believe if you do:
`x = Pipeline -> Instruction -> Pipeline`
Then `x -> Instruction -> Pipeline == x`
Although I wouldn't want to make any guarantees on that.